crates/pallet-evm-balances/src/impl_currency.rs

```diff
@@ -129,14 +131,21 @@ where
             return (NegativeImbalance::zero(), value);
         }
 
-        let result = match Self::try_mutate_account_handling_dust(
+        let mutate_closure = |account: &mut AccountData<<T as Config<I>>::Balance>| -> Result<
```

Comment:
This is quite odd

Reply:
Clippy suggestion to move complex closure used at `try_mutate_account_handling_dust` higher with own `let`. Any advice to do it more better ?

Reply:
```
error: in a `match` scrutinee, avoid complex blocks or closures with blocks; instead, move the block or closure higher and bind it with a `let`
```

Reply:
The issue is at `match Self::try_mutate_account_handling_dust`, so

```rust
let result = Self::try_mutate_account_handling_dust(...);
match result {
```

Reply:
That's what the suggestion from clippy says to do actually.

Reply:
My bad, thanks - fixed